sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java

```diff
@@ -271,25 +270,47 @@ private void loadFallbackValuesFromEnvironment() {
     }
 
     private ArrayList<TokenCredential> getCredentialsChain() {
+        Configuration configuration = identityClientOptions.getConfiguration() == null
+            ? Configuration.getGlobalConfiguration().clone()
+            : identityClientOptions.getConfiguration();
+        String selectedCredentials = configuration.get("AZURE_TOKEN_CREDENTIALS");
+        boolean useProductionCredentials = true;
+        boolean useDeveloperCredentials = true;
+        if (!CoreUtils.isNullOrEmpty(selectedCredentials)) {
+            if ("prod".equalsIgnoreCase(selectedCredentials)) {
+                useDeveloperCredentials = false;
+            } else if ("dev".equalsIgnoreCase(selectedCredentials)) {
+                useProductionCredentials = false;
+            } else {
+                throw LOGGER.logExceptionAsError(new IllegalArgumentException(
+                    "Invalid value for AZURE_TOKEN_CREDENTIALS. Valid values are 'prod' or 'dev'."));
+            }
+        }
+
         ArrayList<TokenCredential> output = new ArrayList<TokenCredential>(8);
-        output.add(new EnvironmentCredential(identityClientOptions.clone()));
-        output.add(getWorkloadIdentityCredential());
-        output.add(new ManagedIdentityCredential(managedIdentityClientId, managedIdentityResourceId, null,
-            identityClientOptions.clone()));
-        output.add(new SharedTokenCacheCredential(null, IdentityConstants.DEVELOPER_SINGLE_SIGN_ON_ID, tenantId,
-            identityClientOptions.clone()));
-        output.add(new IntelliJCredential(tenantId, identityClientOptions.clone()));
-        output.add(new AzureCliCredential(tenantId, identityClientOptions.clone()));
-        output.add(new AzurePowerShellCredential(tenantId, identityClientOptions.clone()));
-        output.add(new AzureDeveloperCliCredential(tenantId, identityClientOptions.clone()));
+        if (useProductionCredentials) {
+            output.add(new EnvironmentCredential(identityClientOptions.clone()));
+            output.add(getWorkloadIdentityCredential());
+            output.add(new ManagedIdentityCredential(managedIdentityClientId, managedIdentityResourceId, null,
+                identityClientOptions.clone()));
+        }
```

Comment:
Should we log when a credential is skipped due to the env var?

Reply:
We're already too noisy so I skipped it. I ended up not doing it because folks with logging enabled will see all the attempted creds anyway and from there can hopefully reason about why something might be missing. I'm happy to add it if we think there's utility.